src/levels.rs

```diff
@@ -390,6 +390,13 @@ impl Neg for Mv {
     }
 }
 
+impl PartialEq for Mv {
+    #[inline(always)]
+    fn eq(&self, other: &Self) -> bool {
+        self.as_bytes() == other.as_bytes()
```

Comment:
Another option to try would be:

```suggestion
        let slf: &[u8; size_of::<Self>()] = zerocopy::transmute_ref!(self);
        let other: &[u8; size_of::<Self>()] = zerocopy::transmute_ref!(other);
        self == other
```

Reply:
If just `.as_bytes()` optimizes well enough, do you think this is necessary?  Or do you think this way, with the array length explicitly known, is more resilient to compiler changes affecting optimization?

Reply:
I think it would also be nice to just test something like:

```rust
(self.x == other.x) & (self.y == other.y)
```

to avoid the branch, it might be that LLVM can optimise this.

Reply:
> If just `.as_bytes()` optimizes well enough, do you think this is necessary? Or do you think this way, with the array length explicitly known, is more resilient to compiler changes affecting optimization?

No clue - just mentioning it since I know that `transmute!` on the values was already tried and decided against, so I wanted to clarify that there are other options you could try that retain the length information at the type level as opposed to relying on the optimizer to deduce the length. Just figured it could be worth testing to see whether it optimizes better than `.as_bytes()`, but I have no clue whether it actually would.

Reply:
Tried this with `Mv` and `RefMvsMvPair` which is the larger struct, and they seem to optimize the same way:

```rust
#[inline(never)]
#[no_mangle]
pub fn mv_eq_as_bytes(mv1: &Mv, mv2: &Mv) -> bool {
    mv1.as_bytes() == mv2.as_bytes()
}

#[inline(never)]
#[no_mangle]
pub fn mv_eq_transmute_ref(mv1: &Mv, mv2: &Mv) -> bool {
    let mv1: &[u8; size_of::<Mv>()] = zerocopy::transmute_ref!(mv1);
    let mv2: &[u8; size_of::<Mv>()] = zerocopy::transmute_ref!(mv2);

    mv1 == mv2
}

#[inline(never)]
#[no_mangle]
pub fn mv_pair_eq_as_bytes(p1: &RefMvsMvPair, p2: &RefMvsMvPair) -> bool {
    p1.as_bytes() == p2.as_bytes()
}

#[inline(never)]
#[no_mangle]
pub fn mv_pair_eq_transmute_ref(p1: &RefMvsMvPair, p2: &RefMvsMvPair) -> bool {
    let p1: &[u8; size_of::<RefMvsMvPair>()] = zerocopy::transmute_ref!(p1);
    let p2: &[u8; size_of::<RefMvsMvPair>()] = zerocopy::transmute_ref!(p2);

    p1 == p2
}
```

```asm
_mv_eq_as_bytes:
Lfunc_begin3:
        .cfi_startproc
        ldr w8, [x0]
        ldr w9, [x1]
        cmp w8, w9
        cset w0, eq
        ret
_mv_eq_transmute_ref:
Lfunc_begin4:
        .cfi_startproc
        ldr w8, [x0]
        ldr w9, [x1]
        cmp w8, w9
        cset w0, eq
        ret
_mv_pair_eq_as_bytes:
Lfunc_begin5:
        .cfi_startproc
        ldr x8, [x0]
        ldr x9, [x1]
        cmp x8, x9
        cset w0, eq
        ret
_mv_pair_eq_transmute_ref:
Lfunc_begin6:
        .cfi_startproc
        ldr x8, [x0]
        ldr x9, [x1]
        cmp x8, x9
        cset w0, eq
        ret
```